EIPS/eip-6366.md

```diff
@@ -0,0 +1,232 @@
+## Motivation
+
+Special roles like `Owner`, `Operator`, `Manager`, `Validator` are common for many smart contracts because permissioned addresses are used to administer and manage them. It is difficult to audit and maintain these system since these permissions are not managed in a single smart contract.
+
+Since permission and role are reflected by the permission token balance of the relevant account in the given ecosystem, cross-interactivity between many ecosystems will be made simpler.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+_Note_ The following specifications use syntax from Solidity `0.8.7` (or above)
+
+### Interface ERC6366 Core
+
+Compliant contracts MUST implement `IERC6366Core`.
+
+```solidity
+interface IERC6366Core {
+  event Transfer(address indexed _from, address indexed _to, uint256 _value);
+
+  event Approval(address indexed _owner, address indexed _delegatee, uint256 _permission);
+
+  function transfer(address _to, uint256 _permission) external returns (bool success);
+
+  function approve(address _delegatee, uint256 _permission) external returns (bool success);
+
+  function permissionOf(address _owner) external view returns (uint256 permission);
+
+  function permissionRequire(uint256 _required, uint256 _permission) external view returns (bool isPermission);
+
+  function delegated(address _owner, address _delegatee) external view returns (uint256 permission);
+}
+```
+
+It is RECOMMENDED to define each permission as a power of `2` so that we can check for the relationship between sets of permissions using the bitwise operator and bitmask.
+
+_Example_
+
+```text
+const PERMISSION_NONE = 0;
+const PERMISSION_READ = 1; // 2⁰
+const PERMISSION_WRITE = 2; // 2¹
+const PERMISSION_EXECUTE = 4; // 2²
+
+// Role admin = 4 | 2 | 1 = 7
+const ROLE_ADMIN = PERMISSION_READ | PERMISSION_WRITE | PERMISSION_EXECUTE;
+// Role operator = 1 | 2 = 3
+const ROLE_OPERATOR = PERMISSION_READ | PERMISSION_WRITE;
+```
+
+The most important permission SHOULD be greater than the lesser one.
+
+#### transfer
+
+Transfers a subset of `_permission` permission to address `_to`, and MUST emit the `Transfer` event.
+The function SHOULD `revert` if the message caller's account permission does not have the subset of the transferring permission. The function SHOULD `revert` if any of transferring permission is existing on target `_to` address.
+
+_Note_ Transfers of `0` permission MUST be treated as normal transfers and emit the `Transfer` event.
+
+```solidity
+function transfer(address _to, uint256 _permission) external returns (bool success);
+```
+
+#### approve
+
+Allows `_delegatee` to act for the permission owner's behalf, up to the `_permission` permission. If this function is called again it overwrites the current granted with `_permission`.
```

Comment:
I think this is missing to remain consistent with the explanations of `transfer`
```suggestion
Allows `_delegatee` to act for the permission owner's behalf, up to the `_permission` permission, and MUST emit the `Approval` event. If this function is called again it overwrites the current granted with `_permission`.
```